python/ray/ml/checkpoint.py

```diff
@@ -99,6 +99,21 @@ class Checkpoint:
             # It is guaranteed that the original data was recovered
             assert isinstance(clf, RandomForestClassifier)
 
+        Checkpoints can be pickled and sent to remote processes.
+        Please note that checkpoints pointing to local directories will be
+        pickled as data representations, so the full checkpoint data will be
+        contained in the checkpoint object. If you want to avoid this,
+        consider passing only the checkpoint directory to the remote task
+        and re-construct your checkpoint object in that function. Note that
+        this will only work if the "remote" task is scheduled on the
```

Comment:
Slightly off the topic as we are mostly talking about using checkpointing across different libraries. But do you foresee us also using this inside of Tune? For example, trial1 saves a checkpoint on node1, which needs to be restored and mutated onto trial2 on node2. In this pbt case, can the driver code be something like 
checkpoint1 = ray_trial_executor.save(trial1) # "serialized" form that basically says you can find it on this path on node2
ray_trial_executor.mutate(trial2, checkpoint1, some_mutation) # which is allocated on node2, which we have no control of.
Do you think Checkpoint impl will be responsible for rsyncing that file from node1 to node2, instead of just throwing errors?
Basically the question is about efficiency and usability - if rsync is more efficient than object store, can user expect Checkpoint impl to rsync files onto different nodes?

Reply:
> Do you think Checkpoint impl will be responsible for rsyncing that file from node1 to node2, instead of just throwing errors?

I don't think Tune should rely on this--- it should directly manipulate the object ref / etc. data in CheckpointManager. The Checkpoint class should only be added at the last minute, when returning the results to the user.